lib/forth_evaluator_web/live/editor.ex

```diff
@@ -0,0 +1,78 @@
+defmodule ForthEvaluatorWeb.Live.Editor do
+  use ForthEvaluatorWeb, :live_view
+
+  alias ForthEvaluator.Repo
+  alias ForthEvaluator.Program
+
+  import Ecto.Query, only: [from: 2]
+
+  def mount(_params, _session, socket) do
+    {:ok,
+     assign(socket,
+       current_program: %Program{text: "", output: ""},
+       program_history: get_program_history()
+     )}
+  end
+
+  def handle_event("submit-program", %{"program-text" => text}, socket) do
+    program =
+      socket.assigns[:current_program]
+      |> Program.changeset(%{text: text, status: "RUNNING"})
+      |> Repo.insert_or_update!()
+
+    socket =
+      socket
+      |> assign(:current_program, program)
+      |> update_program_history()
+
+    start_evaluation(program)
+
+    {:noreply, socket}
+  end
+
+  def handle_event("new-program", _params, socket) do
+    {:noreply, assign(socket, :current_program, %Program{text: "", output: ""})}
+  end
+
+  def handle_event("load-program", %{"id" => id}, socket) do
+    program = Repo.get!(Program, id)
+    {:noreply, assign(socket, :current_program, program)}
+  end
+
+  def handle_info({:evaluation_done, program}, socket) do
+    current_program? = program.id == socket.assigns[:current_program].id
```

Comment:
Remember you can do:
```suggestion
    current_program? = program.id == socket.assigns.current_program.id
```

Reply:
You are right. I updated it in: [Update keyword map access notation](https://github.com/CimimUxMaio/forth-evaluator/pull/10/commits/16c442018b407d62207c0e3a873b74e0e3fb0e5c)